### PR TITLE
pools: Add metrics dataclass

### LINKED ISSUE #135: I want to know what pool a miner is currently active on
In the current MinerData, I can see what the pool configuration is, but if some of those pools are offline, there is no indication of that. Ideally I would like to know the status of every pool in the configuration, but at a minimum, I would like to know which pools are currently active.

### PR DESCRIPTION
**Pull request by Ytemiloluwa**
This Pull Request starts on #135  @b-rowan

### CODE REVIEW

File: pyasic/data/pools.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@dataclass
+class PoolMetrics:
+    """A dataclass to standardize pool metrics returned from miners.
+    Attributes:
+
+    accepted: Number of accepted shares.
+    rejected: Number of rejected shares.
+    get_failures: Number of failures in obtaining work from the pool.
+    remote_failures: Number of failures communicating with the pool server.
+    stratum_active: Indicates if the miner is connected to the stratum server.
```
Comment (b-rowan) [inline pyasic/data/pools.py:13]: This can probably be renamed to just `active` to simplify.

File: pyasic/data/pools.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    pool_rejected_percent: Percentage of rejected shares by the pool.
+    pool_stale_percent: Percentage of stale shares by the pool.
+    """
+
+    accepted: int
+    rejected: int
+    get_failures: int
+    remove_failures: int
+    stratum_active: bool
+
```
Comment (b-rowan) [inline pyasic/data/pools.py:23]: The properties need a `field` here to properly include it into the dataclass.  See these - 
https://github.com/UpstreamData/pyasic/blob/c26b78aa014666ed2cf60123dbaf254b5719a23e/pyasic/data/__init__.py#L104-L106
https://github.com/UpstreamData/pyasic/blob/c26b78aa014666ed2cf60123dbaf254b5719a23e/pyasic/data/__init__.py#L266-L302

**Comment by b-rowan** [comment]
> This Pull Request Fixes #135 @b-rowan

Doesn't fix #135 quite yet, will have to implement the methods for getting the data.

File: pyasic/data/pools.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    active: Indicates if the miner is connected to the stratum server.
+    pool_rejected_percent: Percentage of rejected shares by the pool.
+    pool_stale_percent: Percentage of stale shares by the pool.
+    """
+
+    accepted: int = field(init=False)
+    rejected: int = field(init=False)
+    get_failures: int = field(init=False)
+    remote_failures: int = field(init=False)
+    stratum_active: bool = field(init=False)
```
Comment (b-rowan) [inline pyasic/data/pools.py:22]: Rename this to match docstring?

File: pyasic/data/pools.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
```
Comment (b-rowan) [inline pyasic/data/pools.py:23]: May want to add `alive` here too, a pool can be either `alive` and `active` (currently in use), `alive` and `!active`, (available as a backup), or `!alive` and `!active` (connection failure, may need to be updated).

File: pyasic/data/pools.py (hunk 3 above)
Comment (Ytemiloluwa) [reply pyasic/data/pools.py:22]: you mean `active` instead `stratum_active` ?

Comment (b-rowan) [reply pyasic/data/pools.py:22]: Yes, seems like you only update the docstring, and it isn't consistent with the dataclass attributes anymore.

File: pyasic/data/pools.py (hunk 4 above)
Comment (Ytemiloluwa) [reply pyasic/data/pools.py:23]: Do you mean this?

alive : bool = None
Not_active : bool = None

Comment (b-rowan) [reply pyasic/data/pools.py:23]: Just needs `alive` and `active`, since `not_active = True` is the same thing as `active = False`.

Comment (Ytemiloluwa) [reply pyasic/data/pools.py:23]: Noted.

**Review by b-rowan — APPROVED** [review]

**Pull request merged**